otter/convergence/service.py

```diff
@@ -389,7 +394,14 @@ def converge_one_group(currently_converging, tenant_id, group_id, version,
         # unexpected errors, so convergence will be retried.
         yield err(None, 'converge-non-fatal-error')
     else:
-        if result in (StepResult.FAILURE, StepResult.SUCCESS):
+        if result[0] in (StepResult.FAILURE, StepResult.SUCCESS):
+            # In order to avoid doing extra work and reporting spurious errors,
+            # if the group status is DELETING it means the group has
+            # successfully been deleted by execute_convergence. And so we will
+            # unconditionally delete the divergent flag to avoid any further
+            # queued-up convergences that will imminently fail.
+            if result[1] == ScalingGroupStatus.DELETING:
```

Comment:
Non-blocking:  Is there a case where the group is in deleting but StepResult ends in FAILURE?  If so, how can we recover from this?  Nothing is going to trigger convergence on the group again.  Should we just delete the group anyway?